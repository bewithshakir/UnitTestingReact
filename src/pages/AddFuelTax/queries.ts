import { useMutation, useQuery } from "react-query";
import { AxiosRequestConfig } from "axios";
import axios from "../../infrastructure/ApiHelper";


const addFuelTax = async (payload: any) => {

    const options: AxiosRequestConfig = {
        method: 'post',
        url: '/api/tax-service/fuel-taxes',
        data: payload,
    };
    const { data } = await axios(options);
    return data;
};

export const useAddFuelTax = (onSuccess: any, onError: any) => {
    return useMutation((payload: any) => addFuelTax(payload), {
        onSuccess,
        onError,
        retry: false
    });
};

const fetchSaleTax = async (query: any) => {
    const options: AxiosRequestConfig = {
        method: 'get',
        url: `api/tax-service/fuel-taxes${query}`
    };
    return axios(options);
};

export const useGetFuelTax = (query: any, onSuccess: any, onError: any) => {
    return useQuery(['get-fuel-tax'], () => fetchSaleTax(query), {
        onSuccess,
        onError,
        enabled: !!query,
        retry: false,
    });
};

const editFuelTax = async (payload: any,) => {
    const options: AxiosRequestConfig = {
        method: 'put',
        url: `/api/tax-service/fuel-taxes`,
        data: payload
    };
    const { data } = await axios(options);
    return data;
};

export const useEditFuelTax = (onSuccess: any, onError: any) => {
    return useMutation((payload: any) => editFuelTax(payload), {
        onSuccess,
        onError,
        retry: false
    });
};