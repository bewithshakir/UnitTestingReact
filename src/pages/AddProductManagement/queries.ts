import { useQuery } from "react-query";
import { AxiosRequestConfig } from "axios";
import axios from '../../infrastructure/ApiHelper';

const getProductTypes = async () => {
    const options: AxiosRequestConfig = {
        method: 'get',
        url: '/api/product-service/product/productType?countryCode=us'
    };
    const { data } = await axios(options);
    return data;
};

export const useGetProductTypes = () => {
    return useQuery(["getProductTypes"], () => getProductTypes());
};