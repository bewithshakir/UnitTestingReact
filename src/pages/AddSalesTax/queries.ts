import { useMutation, useQuery } from "react-query";
import { AxiosRequestConfig } from "axios";
import axios from "../../infrastructure/ApiHelper";


const addSalesTax = async (payload: any) => {
    const options: AxiosRequestConfig = {
        method: 'post',
        url: 'api/tax-service/sales-taxes',
        data: payload,
    };
    const { data } = await axios(options);
    return data;
};


export const useAddSalesTax = (onError: any, onSuccess: any) => {
    return useMutation((payload: any) =>
        addSalesTax(payload), {
        onError,
        onSuccess,
        retry: false,
    });
};

// interface PayloadProps {

// }
const fetchSaleTax = async (query: any) => {
    const options: AxiosRequestConfig = {
        method: 'get',
        url: `api/tax-service/sales-taxes/cities${query}`
    };
    return axios(options);
};

export const useGetSaleTax = (query: any, onSuccess: any, onError: any) => {
    return useQuery(['get-sale-tax'], () => fetchSaleTax(query), {
        onSuccess,
        onError,
        enabled: !!query,
        retry: false,
    });
};

const editSaleTax = async (payload: any,) => {
    const options: AxiosRequestConfig = {
        method: 'put',
        url: `/api/tax-service/sales-taxes`,
        data: payload
    };
    const { data } = await axios(options);
    return data;
};

export const useEditSalesTax = (onSuccess: any, onError: any) => {
    return useMutation((payload: any) => editSaleTax(payload), {
        onSuccess,
        onError,
        retry: false
    });
};