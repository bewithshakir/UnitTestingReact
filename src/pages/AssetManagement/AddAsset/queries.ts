import { useMutation, useQuery } from "react-query";
import { AxiosRequestConfig, AxiosResponse } from "axios";
import axios from "../../../infrastructure/ApiHelper";
import AssetManagementModel from "../../../models/AssetManagementModel";


const addDsp = async (payload: AssetManagementModel) => {
    /* const finalPayload = {
        customerId: payload.customerId,
        dspName : payload.dspName,
        contactName: payload.contactNm,
        contactEmailId: payload.email,
        contactPhoneNumber: payload.phone,
        addressLine1: payload.addressLine1,
        addressLine2: payload.addressLine2,
        cityNm: payload.city,
        stateNm: payload.state,
        postalCd: +payload.postalCode
    };
    const options: AxiosRequestConfig = {
        method: 'post',
        url: `api/customer-service/customers/${payload.customerId}/dsps`,
        data: finalPayload,
    };
    const { data } = await axios(options);
    return data; */

    const payloadData = {
        title: 'foo2',
        body: 'bar2',
        userId: 2,
    }
    
    const { data } = await axios.post('https://jsonplaceholder.typicode.com/posts', payloadData);
    return data;
};

export const useAddAsset = (onSuccess: any, onError: any) => {
    return useMutation((payload: AssetManagementModel) =>
    addDsp(payload), {
        onError,
        onSuccess,
        retry: false,
    });
};