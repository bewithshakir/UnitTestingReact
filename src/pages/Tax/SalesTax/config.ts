export const sortByOptions = [
    "taxes.salesTax.sortBy.cityname_atoz",
    "taxes.salesTax.sortBy.cityname_ztoa",
  ];