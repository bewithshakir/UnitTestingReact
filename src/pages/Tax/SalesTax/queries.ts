import { useInfiniteQuery } from "react-query";
import { AxiosRequestConfig } from "axios";
import axios from "../../../infrastructure/ApiHelper";
import { pageDataLimit } from '../../../utils/constants';

const getSalesTaxList = async (pageParam: number, searchTerm: string, sortOrder: any) => {
    const query = new URLSearchParams();
    if (searchTerm) {
        query.append("search", searchTerm);
    }
    
    query.append("sortBy", sortOrder.sortBy);
    query.append("order", sortOrder.order);
    const salesTaxListEntitySet = `/api/tax-service/sales-tax/list?limit=${pageDataLimit}&offset=${pageParam}`;

    const url = query ? `&countryCode=us&${query.toString()}` : `&countryCode=us`;
    const options: AxiosRequestConfig = {
        method: 'get',
        url: salesTaxListEntitySet + url
    };
    const { data } = await axios(options);
    return data;
};

export const salesTaxListSet = (query: string, sortOrder: any) => {
    return useInfiniteQuery(["getSalesTaxList", query, sortOrder], ({ pageParam = 0 }) => getSalesTaxList(pageParam, query, sortOrder), {
        getNextPageParam: (lastGroup: any) => {
          if(lastGroup.data.pagination.offset < lastGroup.data.pagination.totalCount ){
              return lastGroup.data.pagination.offset + 15;
          }
        },
        keepPreviousData: true
    });
};