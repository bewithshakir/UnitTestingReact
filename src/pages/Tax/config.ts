export const sortByOptions = [
  "taxes.fuelTax.sortBy.cityname_atoz",
  "taxes.fuelTax.sortBy.cityname_ztoa",
];