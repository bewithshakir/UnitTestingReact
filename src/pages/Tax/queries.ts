import { useInfiniteQuery } from "react-query";
import { AxiosRequestConfig } from "axios";
import axios from "../../infrastructure/ApiHelper";
import { pageDataLimit } from '../../utils/constants';

const getFuelTaxList = async (pageParam: number, searchTerm: string, sortOrder: { sortBy: string, order: string }, filterParams?: { [key: string]: string[] }) => {
    const query = new URLSearchParams();
    if (searchTerm) {
        query.append("search", searchTerm);
    }
    query.append("sortBy", sortOrder.sortBy);
    query.append("order", sortOrder.order);
    if (filterParams && Object.keys(filterParams).length > 0) {
        for (const key of Object.keys(filterParams)) {
            query.append(key, JSON.stringify(filterParams[key]));
        }
    }

    const fuelTaxListEntitySet = `/api/tax-service/fueltax/list?limit=${pageDataLimit}&offset=${pageParam}`;
    const url = query ? `&countryCode=us&${query.toString()}` : `&countryCode=us`;
    const options: AxiosRequestConfig = {
        method: 'get',
        url: fuelTaxListEntitySet + url
    };
    const { data } = await axios(options);
    return data;
};

export const useFuelTaxList = (query: string, sortOrder: { sortBy: string, order: string }, filterParams?: { [key: string]: string[] }) => {
    return useInfiniteQuery(["getFuelTaxList", query, sortOrder, filterParams], ({ pageParam = 0 }) => getFuelTaxList(pageParam, query, sortOrder, filterParams), {
        getNextPageParam: (lastGroup: any) => {
            if (lastGroup.data.pagination.offset < lastGroup.data.pagination.totalCount) {
                return lastGroup.data.pagination.offset + 15;
            }
        },
        keepPreviousData: true
    });
};