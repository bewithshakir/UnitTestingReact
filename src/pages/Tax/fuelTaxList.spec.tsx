/* eslint-disable no-console */
import { shallow } from 'enzyme';
import TaxModel from '../../models/TaxModel';
import ActionsMenu from '../../components/UIComponents/Menu/ActionsMenu.component';
import SortbyMenu from '../../components/UIComponents/Menu/SortbyMenu.component';
import { FuelTax } from './config';
import { act, renderHook } from "@testing-library/react-hooks";
import { rest } from "msw";
import { serverMsw } from "../../setupTests";
import { createWrapper } from "../../tests/utils";
// import FuelTaxList from './index';
import { useFuelTaxList } from './queries';


jest.mock("react-query", () => {
    return {
        useInfiniteQuery: jest.fn(),
    };
});
jest.mock("./queries", () => {
    return {
        useFuelTaxList: jest.fn(),
    };
});

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockedUsedNavigate,
}));


describe('Given Bruger Menu on FuelTax Landing Page', () => {
    const TaxObj = new TaxModel();
    const massActionOptions = TaxObj.massActions();
    test('Render Bruger Menu', () => {
        const FuelTaxBurgerMenu = shallow(
            <ActionsMenu
                options={massActionOptions}
                onSelect={() => jest.fn()}
            />
        );
        expect(FuelTaxBurgerMenu).toMatchSnapshot();
    });

    test('Bruger Menu Menu With Options', () => {
        const FuelTaxBurgerMenu = shallow(
            <ActionsMenu
                options={massActionOptions}
                onSelect={() => jest.fn()}
            />
        );
        FuelTaxBurgerMenu.find(".btn-listmemu").simulate('click');
        expect(FuelTaxBurgerMenu.find(".btn-listmemu").hasClass('active')).toBe(true);
        expect(FuelTaxBurgerMenu.find('.actions-popper').exists()).toBe(true);
    });

});

describe('Given Sortby Menu on FuelTax Landing Page-', () => {
    const { SortByOptions } = FuelTax.LandingPage;
    test('Render Sortby Menu', () => {
        const FuelTaxSortbyMenu = shallow(
            <SortbyMenu
                options={SortByOptions.map((sortByItem) => (sortByItem))}
                onSelect={() => jest.fn()}
            />
        );
        expect(FuelTaxSortbyMenu).toMatchSnapshot();
    });

    test('Sortby Menu Menu With Options', () => {
        const FuelTaxSortbyMenu = shallow(
            <SortbyMenu
                options={SortByOptions.map((sortByItem) => (sortByItem))}
                onSelect={() => jest.fn()}
            />
        );
        FuelTaxSortbyMenu.find(".btn-sortby").simulate('click');
        expect(FuelTaxSortbyMenu.find(".btn-sortby").hasClass('active')).toBe(true);
        expect(FuelTaxSortbyMenu.find('.sortby-popper').exists()).toBe(true);
    });

});

describe('GetFuelTaxList', () => {
    it('Success returns data', async () => {
        const { result, waitFor } = renderHook(() => useFuelTaxList('', { sortBy: "", order: "" }, {}), {
            wrapper: createWrapper()
        });
        act(() => {
            result.current.fetchNextPage();
        });
        await waitFor(() => {
            return result.current.isSuccess;
        });
        expect(result.current.status).toBe('success');
    });

    it('fail to get data', async () => {
        serverMsw.use(
            rest.get('*', (req, res, ctx) => res(ctx.status(500)))
        );
        const { result, waitFor } = renderHook(() => useFuelTaxList('9999', { sortBy: "", order: "" }, {}), {
            wrapper: createWrapper()
        });
        act(() => {
            result.current.fetchNextPage();
        });

        await waitFor(() => result.current.isError);
        expect(result.current.error).toBeDefined();
    });
});

