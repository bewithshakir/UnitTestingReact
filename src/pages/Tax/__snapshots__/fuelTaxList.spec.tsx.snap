// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Bruger Menu on FuelTax Landing Page Render Bruger Menu 1`] = `
<div>
  <WithStyles(ForwardRef(Button))
    aria-haspopup="true"
    aria-label="actions menu list"
    className="btn-listmemu "
    onClick={[Function]}
    size="small"
    startIcon={
      <WithStyles(ForwardRef(SvgIcon))
        component={[Function]}
        viewBox="0 0 20 20"
      />
    }
    variant="contained"
  />
  <ForwardRef(Popper)
    anchorEl={null}
    className="actions-popper"
    disablePortal={true}
    open={false}
    placement="bottom-end"
    transition={true}
  >
    <Component />
  </ForwardRef(Popper)>
</div>
`;

exports[`Given Sortby Menu on FuelTax Landing Page- Render Sortby Menu 1`] = `
<div>
  <WithStyles(ForwardRef(Button))
    aria-haspopup="true"
    aria-label="sortby menu list"
    className="btn-sortby "
    onClick={[Function]}
    startIcon={<Memo />}
  >
    buttons.sort by
  </WithStyles(ForwardRef(Button))>
  <ForwardRef(Popper)
    anchorEl={null}
    className="sortby-popper"
    disablePortal={true}
    open={false}
    placement="bottom-start"
    transition={true}
  >
    <Component />
  </ForwardRef(Popper)>
</div>
`;
